Classes/Command/FluidTemplatePrecompileCommand.php

```diff
@@ -0,0 +1,163 @@
+        $limit = $input->getOption('limit');
+        $disableErrorHandlers = $input->getOption('disable-error-handler');
+
+        // Test exception handlers
+        if (set_error_handler(function() {}) && !$disableErrorHandlers) {
+            $outputHelper->warning(
+                "\nIt seems like there is an error handler registered."
+                . "\nIn some cases this can prevent the precompiling to run properly."
+                . "\nAn ErrorHandler might raise an exception for a PHP Warning."
+            );
+
+            restore_error_handler();
+        }
+
+        $result = $this->getPrecompilerService()->warmup($extension);
+
+        // General information
+        $templatesCount = 0;
+        foreach ($result as $extensionResult) {
+            $templatesCount += count($extensionResult);
+        }
+
+        $uncompilable = array_column($result, 'uncompilable');
+        $uncompilable = array_sum($uncompilable);
+
+        $outputHelper->section('General');
+        $outputHelper->text(vsprintf('Compiled %d extension(s) with %d template(s) in total.', [count($result), $templatesCount]));
+        $outputHelper->text(vsprintf('%d template(s) could not be compiled.', [$uncompilable]));
+
+        // Additional information
+        if ($output->isVerbose()) {
+            foreach ($result as $extensionKey => $extensionResult) {
+                $templates = $extensionResult['results'];
+
+                if ($onlyFailed) {
+                    $templates = array_filter($templates, function ($template) {
+                        return false == $template[FluidCacheWarmupResult::RESULT_COMPILABLE];
+                    });
+                }
+
+                $templates = array_slice($templates, 0, $limit);
+
+                $templateRows = [];
+                foreach ($templates as $templateName => $templateAttributes) {
+                    $templateRows[] = ['Template', $templateName];
+
+                    foreach ($templateAttributes as $attributeName => $attributeValue) {
+                        $templateRows[] = [
+                            $attributeName,
+                            is_array($attributeValue) ? implode(PHP_EOL, $attributeValue) : static::formatReadable($attributeValue)
+                        ];
+                    }
+
+                    $templateRows[] = new TableSeparator();
+                }
+
+                array_pop($templateRows);
+
+                $outputHelper->section(sprintf('Extension: %s (%d uncompilable of %d)', $extensionKey, $extensionResult['uncompilable'], count($templates)));
+
+                $outputHelper->table(
+                    [],
+                    $templateRows
+                );
+            }
+        }
+
+        if ($disableErrorHandlers) {
+            restore_error_handler();
+        }
+
+        if ($fail && $uncompilable > 0) {
+            throw new \Exception(sprintf('Could not compile %d templates.', $uncompilable), 1518205183);
```

Comment:
Please use `return` instead here.

Reply:
Do you mean to remove the exception and return an invalid status code or to return the exception?
Latter would be wrong because the Command's `execute()` function should only return a status code.

I'd prefer a thrown exception because it is more highlighted in a console. A simple non-zero return code might not be as noticeable as a big red box.

Reply:
Failure in this CLI task is by definition supposed to be an error return code, not a runtime exception. The reason behind that is that an exception is indistinguishable from an internal error, whereas a good old return code communicates exactly what it should: that the task failed because you told it to fail if it couldn't compile.

However, a message to go along with that return code certainly wouldn't hurt.